### PR TITLE
RavenDB-12184: Check if there is a ConnectionString before adding rav…

### PR DESCRIPTION
**Pull request by garayx**
…en/sql ETL command

### CODE REVIEW

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 1]
```diff
@@ -1029,6 +1029,23 @@ private void UpdateDatabase(TransactionOperationContext context, string type, Bl
                         }
 
                         databaseRecord = JsonDeserializationCluster.DatabaseRecord(databaseRecordJson);
+                        
+                        if (type == "AddRavenEtlCommand" || type == "AddSqlEtlCommand")
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs]: This should be handled when you are putting new ETL. In `OutgoingTasksHandler.AddEtl`

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: Also I think that when we want to remove connection string in `OutgoingTasksHandler.RemoveConnectionString` then we need to check if any ETLs are using it or not and throw exception with list of them when we find any.

Comment (garayx) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: done

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 2]
```diff
@@ -1324,13 +1324,19 @@ public async Task<(long, object)> ModifyPeriodicBackup(TransactionOperationConte
...
             UpdateDatabaseCommand command;
+            var databaseRecord = LoadDatabaseRecord(databaseName, out var _);
+            var connectionStringName = etlConfiguration["ConnectionStringName"].ToString();
 
             switch (EtlConfiguration<ConnectionString>.GetEtlType(etlConfiguration))
             {
                 case EtlType.Raven:
+                    if (databaseRecord.RavenConnectionStrings.TryGetValue(connectionStringName, out var rvnConnection) == false)
+                        ThrowUnknownConnectionStringName(connectionStringName);
                     command = new AddRavenEtlCommand(JsonDeserializationCluster.RavenEtlConfiguration(etlConfiguration), databaseName);
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/ServerStore.cs]: We are deserializing the configuration here (`JsonDeserializationCluster.RavenEtlConfiguration(etlConfiguration)`). Use this instance and:
1. Fire `.Validate()` method and throw if there are any errors
2. Add new method in `abstract EtlConfiguration` called `.ValidateConnectionString(DatabaseRecord)`

**Review by ppekrol — APPROVED** [review]

**Pull request merged**